core/tests/webdriverio_utils/waitFor.js

```diff
@@ -204,6 +204,19 @@ var numberOfElementsToBe = async function(
   });
 };
 
+var numberOfChainedElementsToBe = async function(
+    parentSelector, childSelector, elementName, number) {
+  await browser.waitUntil(async function() {
+    var element = await $(parentSelector).$$(childSelector);
+    return element.length === number;
+  },
+  {
+    timeout: DEFAULT_WAIT_TIME_MSECS,
+    timeoutMsg: `Number of ${elementName} is not equal to ${number}\n` +
+    new Error().stack + '\n'
+  });
+};
```

Comment:
I got another idea for this which might reduce a bit of duplicacy. That is to use this in [noOfElements](https://github.com/oppia/oppia/blob/develop/core/tests/webdriverio_utils/waitFor.js#L194) only by taking 4 parameters(parentSelector, childSelector, elementName, number) in noOfElements and setting childSelector to be null intially in the parameter itself. Now if the child selector has a value not equal to null then you can use the logic of chained selectors else use the logic of single selectors.

The reason I am telling this because `numberOfChainedElementsToBe` doesn't seem to express the meaning of this function. Also, as discussed in our prev meeting we couldn't come to a better name too.

Reply:
Done.